src/random.h

```diff
@@ -81,6 +81,16 @@ class FastRandomContext {
         }
     }
 
+    unsigned long randrange(unsigned long range)
+    {
+        --range;
+        int bits = 8 * sizeof(long) - __builtin_clzl(range);
```

Comment:
CLZ is undefined for 0, a range of 1 is dumb but might be mechanically generated in some case.  At a minimum there should be a comment that range must be greater than 1.